addons/website/static/src/scss/website.scss

```diff
@@ -1698,6 +1698,12 @@ input[value*="data-oe-translation-initial-sha"] {
     --Notification__background-color: #{$body-bg};
 }
 
+// Handle style of shared OWL components between back-end and front-end
+.modal.o_technical_modal {
+    --Modal__header-background: #{$body-bg};
+    --Modal__header-color: #{$body-color};
```

Comment:
I added these variables because at the moment, the modal-header of the `Dialog` component inherits the color of the navbar of the backend. Example here on master in community. Our UI elements have the primary of our color-palette but the header has the background-color of the navbar of community: 

<img width="160" alt="Screenshot 2024-06-07 at 11 33 58" src="https://github.com/odoo-dev/odoo/assets/110090660/8caf5289-01a7-4e81-a450-7d079c2d4345">

In enterprise it's less noticeable as the navbar is white but if you switch to a dark background, you will see it.

These variables allow to keep this behaviour in the back-end but will tweak it in the front-end to avoid this issue. 

> My guess is that remove the lines currently in master between 52 to 56 should be enough, could you check if true and why these are there in the first place?

If we do that, it will indeed work fine in enterprise and in the front-end (it will be the same as what you sent as screenshots) **but** the modals in community backend will lose their colored background. I don't think that's what we want ?

Reply:
Oh.. community...  indeed 😕
I'll test the behaviour again after the rebase, but your approach seems reasonable indeed 

Thanks for the explanation!